pkg/reconciler/gc/v2/gc_test.go

```diff
@@ -207,6 +207,27 @@ func TestCollect(t *testing.T) {
 				WithCreationTimestamp(oldest),
 				WithLastPinned(tenMinutesAgo)),
 		},
+	}, {
+		name: "keep oldest because of the preserve annotation",
+		cfg: cfg("keep-all", "foo", 5556,
+			WithLatestCreated("5556"),
+			WithLatestReady("5556"),
+			WithConfigObservedGen),
+		revs: []*v1.Revision{
```

Comment:
We might want a unit test that does delete one of them:

eg.
1. Oldest, preserved
2. Old enough for deletion, gets deleted
3. Not old enough

and see that it does delete 2

Reply:
Done.